File: src/Crawler/Result/Storage/Storage.php
<?php

/*
 * This file is part of the SiteOne Crawler.
 *
 * (c) Ján Regeš <jan.reges@siteone.cz>
 */

declare(strict_types=1);

namespace Crawler\Result\Storage;

interface Storage
{
    public function save(string $uqId, string $content): void;

    public function load(string $uqId): string;

    public function delete(string $uqId): void;

    public function deleteAll(): void;
}

File: src/result/storage/storage.rs
// SiteOne Crawler - Storage trait
// (c) Jan Reges <jan.reges@siteone.cz>

use crate::error::CrawlerResult;

pub trait Storage: Send + Sync {
    fn save(&mut self, uq_id: &str, content: &[u8]) -> CrawlerResult<()>;

    fn load(&self, uq_id: &str) -> CrawlerResult<Vec<u8>>;

    fn delete(&mut self, uq_id: &str) -> CrawlerResult<()>;

    fn delete_all(&mut self) -> CrawlerResult<()>;
}
